### ISSUE TITLE
Unused variable false positive inside array_reduce

### ISSUE THREAD

**Issue by calexandrepcjr** [slevomat/coding-standard#533]
Slevomat claims that $query is not being used in this piece of code, but it is, it's just the array_reduce buffer.

----------------------------------------------------------------------                                               
FOUND 1 ERROR AFFECTING 1 LINE                                                                                       
----------------------------------------------------------------------                                               
 109 | ERROR | Unused variable $query.                                                                               
     |       | (SlevomatCodingStandard.Variables.UnusedVariable.UnusedVariable)                                      
----------------------------------------------------------------------  

```
$rows = array_reduce(
                $values,
                static function ($query, $row) {
                    if ($query !== null) {
                        $query .= ',';
                    }

                    return  $query .= sprintf(' (%s)', implode(', ', $row));
                }
            );
```
Look that not even with a reference function the slevomat behaves correctly, throwing the same error to me:

```
        foreach ($data as $tableName => $values) {
            $rows = array_reduce(
                $values,
                ['self', 'generateValues']
            );
        }

    protected static function generateValues(string $query, array $row): string
    {
        if ($query !== null) {
            $query .= ',';
        }

        return $query .= sprintf(' (%s)', implode(', ', $row));
    }
```

**Comment by ondrejmirtes** [comment]
I'd update this:

```php
return  $query .= sprintf(' (%s)', implode(', ', $row));
```

to this:

```php
return  $query . sprintf(' (%s)', implode(', ', $row));
```

**Comment by calexandrepcjr** [comment]
Seems reasonable, @ondrejmirtes , in my case I put inside the `sprintf('%s (%s)', $query, implode(', ', $row));`, but the error intrigued me.
